src/main/kotlin/uk/gov/justice/digital/hmpps/deliusapi/service/contact/ContactService.kt

```diff
@@ -363,12 +364,19 @@ class ContactService(
     val sortOrder = if (request.sortDirection == ASCENDING) SortOrder.ASC else SortOrder.DESC
 
     // Set up sort type defaulting to score first then lastUpdatedDateTime unless specified
-    searchResponseBuilder = if (request.sort == LAST_UPDATED_DATETIME) {
-      searchResponseBuilder.withSort(fieldSort("last_updated_date_time").order(sortOrder))
-      searchResponseBuilder.withSort(fieldSort("_score").order(sortOrder))
-    } else {
-      searchResponseBuilder.withSort(fieldSort("_score").order(sortOrder))
-      searchResponseBuilder.withSort(fieldSort("last_updated_date_time").order(sortOrder))
+    searchResponseBuilder = when (request.sort) {
+      LAST_UPDATED_DATETIME -> {
+        searchResponseBuilder.withSort(fieldSort("last_updated_date_time").order(sortOrder))
+        searchResponseBuilder.withSort(fieldSort("_score").order(sortOrder))
+      }
+      MOST_RECENT_DATE_FIRST -> {
+        searchResponseBuilder.withSort(fieldSort("contact_date.date").order(sortOrder))
+        searchResponseBuilder.withSort(fieldSort("_score").order(sortOrder))
+      }
+      else -> {
+        searchResponseBuilder.withSort(fieldSort("_score").order(sortOrder))
+        searchResponseBuilder.withSort(fieldSort("last_updated_date_time").order(sortOrder))
+      }
     }
```

Comment:
Could this be simplified?  Would something like this work:
```suggestion
    searchResponseBuilder = when (request.sort) {
      LAST_UPDATED_DATETIME -> searchResponseBuilder.withSort(fieldSort("last_updated_date_time").order(sortOrder))
      MOST_RECENT_DATE_FIRST -> searchResponseBuilder.withSort(fieldSort("contact_date.date").order(sortOrder))
    }
    searchResponseBuilder.withSort(fieldSort("_score").order(sortOrder))
```